go/mysql/mysql56_gtid_set.go

```diff
@@ -407,6 +473,32 @@ func NewMysql56GTIDSetFromSIDBlock(data []byte) (Mysql56GTIDSet, error) {
 	return set, nil
 }
 
+// popInterval will look at the two pre-sorted interval stacks supplied, and if at least one of the stacks is non-empty
+// will mutate the destination interval with the next earliest interval based on start sequence.
+// popInterval will return true if we were able to pop, or false if both stacks are now empty.
+func popInterval(dst *interval, s1, s2 *[]interval) bool {
+	if s1 == nil || s2 == nil {
```

Comment:
When a function doesn't admit the possibility of failure (no `error` return), I think it's better to let the panic happen rather than silently continue if failure nevertheless occurs. It's a bug if anything calls this with a nil pointer, and a panic makes it very easy to detect and fix that bug.

Reply:
I think that's a good point. The reason I choose to return false is because I see the functions return as an answer to the question, "were you able to successfully pop an interval?" The answer in this case is no, we were not successful. I agree though that it silently hides the error, so I'll remove this conditional check.

Reply:
I think I'd make a subtle distinction that the question callers expect to be answered is more like: "Is there a next item to process?" If one of these inputs is invalid, the actual answer to this question is neither true nor false but rather, "You haven't given me enough information to answer that question."

It's like expecting a numerical answer to, "How long is a piece of string?" The real answer is `panic("you haven't told me what string you're talking about")`.

Reply:
Updated: https://github.com/vitessio/vitess/pull/6180/commits/cfeb086036fd857c27c81e9722e366a93009426b.